From 71839bcbc11135b828b377cdce5fdfd5b096239b Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Tue, 4 Aug 2026 20:03:11 -0400
Subject: [PATCH 263/268] kernel32: Add stub for IsNativeVhdBoot.

EpicNu seems to require it (along with PROTON_USE_WOW64=1)
---
 dlls/kernel32/kernel32.spec |  1 +
 dlls/kernel32/kernel_main.c | 11 +++++++++++
 2 files changed, 12 insertions(+)

diff --git a/dlls/kernel32/kernel32.spec b/dlls/kernel32/kernel32.spec
index 1a3f5576f6d..1e007bde175 100644
--- a/dlls/kernel32/kernel32.spec
+++ b/dlls/kernel32/kernel32.spec
@@ -1010,6 +1010,7 @@
 @ stdcall -import IsDBCSLeadByteEx(long long)
 @ stdcall -import IsDebuggerPresent()
 @ stub -i386 IsLSCallback
+@ stdcall IsNativeVhdBoot(ptr)
 @ stdcall -import IsNLSDefinedString(long long ptr wstr long)
 @ stdcall -import IsNormalizedString(long wstr long)
 @ stdcall -import IsProcessInJob(long long ptr)
diff --git a/dlls/kernel32/kernel_main.c b/dlls/kernel32/kernel_main.c
index edf54e84b72..54bb0e67b09 100644
--- a/dlls/kernel32/kernel_main.c
+++ b/dlls/kernel32/kernel_main.c
@@ -223,3 +223,14 @@ HANDLE WINAPI CreateBoundaryDescriptorA(LPCSTR name, ULONG flags)
     FIXME("%s, %#lx - stub\n", debugstr_a(name), flags);
     return NULL;
 }
+
+/******************************************************************************
+ *          IsNativeVhdBoot    (KERNEL32.@)
+ */
+BOOL WINAPI IsNativeVhdBoot(BOOL *native)
+{
+    FIXME("%p stub!\n", native);
+
+    if (native) *native = FALSE;
+    return FALSE;
+}
-- 
2.55.0

